addons/l10n_pt_pos/tests/test_l10n_pt_pos.py

```diff
@@ -0,0 +1,97 @@
+
+        # The following field is not part of the hash so it can be modified
+        order.note = 'new note'
+
+    def test_l10n_pt_pos_document_no(self):
+        """
+        Test that the document number for Portugal follows this format: [^ ]+ [^/^ ]+/[0-9]+
+        """
+        for expected in ['pos_order PoS_Shop_Test/0001', 'pos_order PoS_Shop_Test/0002', 'pos_order PoS_Shop_Test/0003']:
+            order = self._create_pos_order()
+            self.assertEqual(order.l10n_pt_pos_document_number, expected)
+
+    def test_l10n_pt_pos_hash_integrity_report(self):
+        """Test the hash integrity report"""
+        order1 = self._create_pos_order("2023-01-01")
+        self._create_pos_order("2023-01-02")
+        order3 = self._create_pos_order("2023-01-03")
+        order4 = self._create_pos_order("2023-01-04")
+        self.assertEqual(order1.l10n_pt_pos_inalterable_hash, False)
+        self.env['pos.order'].l10n_pt_pos_compute_missing_hashes(order1.company_id.id)  # Called when printing the receipt
+
+        integrity_check = next(filter(lambda r: r['name'] == order1.config_id.name, self.company_pt._l10n_pt_pos_check_hash_integrity()['results']))
+        self.assertEqual(integrity_check['status'], 'verified')
+        self.assertEqual(integrity_check['msg'], 'Entries are correctly hashed')
+        self.assertEqual(integrity_check['from_date'], fields.Date.to_string(order1.date_order))
+        self.assertEqual(integrity_check['to_date'], fields.Date.to_string(order4.date_order))
+
+        # Let's change one of the fields used by the hash. It should be detected by the integrity report.
+        # We need to bypass the write method of pos.order to do so.
+        Model.write(order3, {'date_order': fields.Date.from_string('2022-01-07')})
+        integrity_check = next(filter(lambda r: r['name'] == order1.config_id.name, self.company_pt._l10n_pt_pos_check_hash_integrity()['results']))
+        self.assertEqual(integrity_check['status'], 'corrupted')
+        self.assertEqual(integrity_check['msg'], f'Corrupted data on pos.order with id {order3.id} ({order3.name}).')
+
+        # Let's try with the l10n_pt_pos_inalterable_hash field itself
+        Model.write(order3, {'date_order': fields.Date.from_string("2023-01-03")})  # Revert the previous change
+        Model.write(order4, {'l10n_pt_pos_inalterable_hash': '$1$fake_hash'})
+        integrity_check = next(filter(lambda r: r['name'] == order1.config_id.name, self.company_pt._l10n_pt_pos_check_hash_integrity()['results']))
+        self.assertEqual(integrity_check['status'], 'corrupted')
+        self.assertEqual(integrity_check['msg'], f'Corrupted data on pos.order with id {order4.id} ({order4.name}).')
+
+    def test_l10n_pt_pos_dont_hash_linked_stock_pickings(self):
+        """Test that creating a POS order (which creates a stock picking) doesn't hash this created stock picking"""
```

Comment:
If you don't hash it, does it mean that you can still modify it?
And so make inconsistencies with the PoS order?